lib/rules/spaced-comment.js

```diff
@@ -281,26 +328,36 @@ module.exports = {
 
             // Checks.
             if (requireSpace) {
-                if (!rule.regex.test(node.value)) {
+                if (!rule.beginRegex.test(node.value)) {
```

Comment:
Just a thought, but is it possible to just do
```
beginMatched = rule.beginRegex.exec(node.value);
endMatched = rule.endRegex.exec(node.value);
``` 
at the top right after the early return and then use those two values for all the subsequent checks? Right now it's also using `.test()` in addition to `exec()`, but it seems like you need to calculate both whether `requireSpace` is `true` or not. 

Could make the code clearer and read a bit nicer (above could become `if (!beginMatched)`, and so on. Just a thought!